servers/physics_2d/godot_area_2d.cpp

```diff
@@ -78,13 +78,6 @@ void GodotArea2D::set_space(GodotSpace2D *p_space) {
 }
 
 void GodotArea2D::set_monitor_callback(const Callable &p_callback) {
-	ObjectID id = p_callback.get_object_id();
-
-	if (id == monitor_callback.get_object_id()) {
```

Comment:
This looks like some optimization and should work without an object too (though the condition probably needs changing). Why was this removed?

Reply:
For object-less callables this will always fall into the optimized path because `get_object_id()` will always return `ObjectID(0)`, so for those callables its no good, and why I removed it. Im not sure what this condition could be changed to.

I tried to see what this optimization and the code that follows was about, but I'm not so familiar with the physics servers inner workings. This code goes all they way back to the original opensource commit, so no explanation there. When I tried to follow all the paths that could lead here it seemed that the optimized path doesn't ever get used in the current code, tho I didn't test it.

In short, I don't know why that code is there in the first place, how to properly change it to support object-less callables (tho maybe just adding `p_callback.is_standard() && ` to the condition would be enough?), and it doesn't appear that removing it would harm anything.

Reply:
Maybe something like `id.is_valid() && id == monitor_callback.get_object_id()`?
This code allows to swap callbacks without clearing shapes if the object is the same. So the condition I suggested would make it skipped if the new callback has no object.

That said, I'm also not very familiar with physics code either. This "optimization" doesn't look very important; normally the callbacks are changed only when toggling area's `monitoring` property, which is rather uncommon thing to do.